src/overlays/actors/ovl_En_Js/z_en_js.c

```diff
@@ -28,89 +39,1021 @@ const ActorInit En_Js_InitVars = {
+
+static f32 D_8096ABF4[] = { 60.0f, 0.0f, 0.0f, 0.0f, 0.0f };
+
+static f32 D_8096AC08[] = { -450.0f, 1400.0f, 1470.0f, 1670.0f, 1470.0f };
+
+static f32 D_8096AC1C[] = { 0.0f, 700.0f, 900.0f, 900.0f, 900.0f };
+
+static Vec3f D_8096AC30 = { 500.0f, -500.0f, 0.0f };
+
+void EnJs_Init(Actor* thisx, PlayState* play) {
+    s32 pad;
+    s16 cs;
+    s32 i;
+    EnJs* this = THIS;
+
+    Actor_SetScale(&this->actor, 0.01f);
+    ActorShape_Init(&this->actor.shape, 0.0f, ActorShadow_DrawCircle, 20.0f);
+    SkelAnime_InitFlex(play, &this->skelAnime, &gMoonChildSkel, &gMoonChildStandingAnim, this->jointTable,
+                       this->morphTable, MOONCHILD_LIMB_MAX);
+    Animation_PlayLoop(&this->skelAnime, &gMoonChildStandingAnim);
+    Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+
+    this->actionFunc = func_8096971C;
+    this->maskType = 0;
+    this->actor.terminalVelocity = -9.0f;
+    this->actor.gravity = -1.0f;
+
+    cs = this->actor.cutscene;
+    i = 0;
+    while (i != 2) {
+        this->cutscenes[i] = cs;
+        if (cs != -1) {
+            this->actor.cutscene = cs;
+            cs = ActorCutscene_GetAdditionalCutscene(this->actor.cutscene);
+        }
+        i++;
+    }
+
+    this->cutsceneIndex = -1;
+
+    switch (ENJS_GET_TYPE(&this->actor)) {
+        case 0:
+            this->maskType = 0;
+            this->actionFunc = func_8096A6F4;
+
+            Animation_PlayLoop(&this->skelAnime, &gMoonChildSittingAnim);
+            func_8016566C(0x3C);
+
+            if (gSaveContext.save.weekEventReg[84] & 0x20) {
+                Inventory_DeleteItem(ITEM_MASK_FIERCE_DEITY, SLOT(ITEM_MASK_FIERCE_DEITY));
+                gSaveContext.save.weekEventReg[84] &= (u8)~0x20;
+                break;
+            }
+        default:
+            break;
+        case 1:
+        case 2:
+        case 3:
+        case 4:
+            this->maskType = ENJS_GET_TYPE(&this->actor);
+            func_80968B8C(this, play);
+            this->actionFunc = func_80969B5C;
+            func_80968A5C(this);
+            if (func_809692A8(ENJS_GET_TYPE(&this->actor) + 4)) {
+                Actor_MarkForDeath(&this->actor);
+                break;
+            }
+            break;
+        case 5:
+        case 6:
+        case 7:
+        case 8:
+            this->maskType = ENJS_GET_TYPE(&this->actor) - 4;
+            this->actionFunc = func_8096A104;
+            break;
+    }
+}
+
+void EnJs_Destroy(Actor* thisx, PlayState* play) {
+    u32 paramsF;
+    EnJs* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+
+    paramsF = ENJS_GET_TYPE(&this->actor);
+    switch (paramsF) {
+        case 0:
+            func_80165690();
+            return;
+        case 5:
+        case 6:
+        case 7:
+        case 8:
+            if (!func_809692A8(paramsF)) {
+                func_80969400(ENJS_GET_TYPE(&this->actor));
+            }
+        default:
+            return;
```

Comment:
```suggestion
            break;
        case 5:
        case 6:
        case 7:
        case 8:
            if (!func_809692A8(paramsF)) {
                func_80969400(ENJS_GET_TYPE(&this->actor));
            }
            break;
        default:
            break;
```
These should be breaks. Also, if it falls through to an immediate break, you can probably put a break in the cases 5/6/7/8